src/gt4py/next/iterator/transforms/global_tmps.py

```diff
@@ -483,19 +530,25 @@ def update_domains(node: FencilWithTemporaries, offset_provider: Mapping[str, An
                         nbt_provider = offset_provider[offset_name]
                         old_axis = nbt_provider.origin_axis.value
                         new_axis = nbt_provider.neighbor_axis.value
-                        consumed_domain.ranges.pop(old_axis)
```

Comment:
Unrelated to this PR. This approach of popping from the dict failed when the domain had more than one axis as the order was not preserved.

Reply:
why does the order matter?